### PR TITLE
Update value of driver name in client metadata to reflect new structure

### PR DESCRIPTION
**Pull request by jyemin**
* mongo-java-driver|sync: for com.mongodb.MongoClient
* mongo-java-driver|legacy: for com.mongodb.client.MongoClient
* mongo-java-driver|reactive-streams: for com.mongodb.reactivestreams.client.MongoClient
* mongo-java-driver|reactive-streams|scala for org.mongodb.scala.MongoClient

https://jira.mongodb.org/browse/JAVA-3488

### CODE REVIEW

File: driver-core/src/main/com/mongodb/internal/async/client/MongoClients.java [hunk 1]
```diff
@@ -149,11 +149,9 @@ private static Cluster createCluster(final MongoClientSettings settings,
                                          @Nullable final MongoDriverInformation mongoDriverInformation,
                                          final StreamFactory streamFactory, final StreamFactory heartbeatStreamFactory) {
         notNull("settings", settings);
-        MongoDriverInformation.Builder builder = mongoDriverInformation == null ? MongoDriverInformation.builder()
-                : MongoDriverInformation.builder(mongoDriverInformation);
         return new DefaultClusterFactory().createCluster(settings.getClusterSettings(), settings.getServerSettings(),
                 settings.getConnectionPoolSettings(), streamFactory, heartbeatStreamFactory, settings.getCredential(),
-                getCommandListener(settings.getCommandListeners()), settings.getApplicationName(), builder.driverName("async").build(),
+                getCommandListener(settings.getCommandListeners()), settings.getApplicationName(), mongoDriverInformation,
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/async/client/MongoClients.java]: Remove "async" as a reported driver name

File: driver-legacy/src/examples/tour/Decimal128LegacyAPIQuickTour.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
      *
      * @param args takes an optional single argument for the connection string
      */
-    public static void main(final String[] args) {
+    public static void main(final String[] args) {
```
Comment (jyemin) [inline driver-legacy/src/examples/tour/Decimal128LegacyAPIQuickTour.java]: Whoops.  I'll revert this.

File: driver-legacy/src/main/com/mongodb/MongoClient.java [hunk 3]
```diff
@@ -732,10 +732,15 @@ private static Cluster createCluster(final ClusterSettings clusterSettings, @Nul
                 credential,
                 getCommandListener(options.getCommandListeners()),
                 options.getApplicationName(),
-                mongoDriverInformation,
+                wrapMongoDriverInformation(mongoDriverInformation),
                 options.getCompressorList());
     }
 
+    private static MongoDriverInformation wrapMongoDriverInformation(@Nullable final MongoDriverInformation mongoDriverInformation) {
+        return (mongoDriverInformation == null ? MongoDriverInformation.builder() : MongoDriverInformation.builder(mongoDriverInformation))
```
Comment (jyemin) [inline driver-legacy/src/main/com/mongodb/MongoClient.java]: This fixes is a regression introduced when I merged the Mongo class into MongoClient

File: driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/MongoClients.java [hunk 4]
```diff
@@ -112,6 +113,11 @@ private static MongoClient create(final com.mongodb.internal.async.client.MongoC
         return new MongoClientImpl(asyncMongoClient);
     }
 
+    private static MongoDriverInformation wrapMongoDriverInformation(final MongoDriverInformation mongoDriverInformation) {
+        return (mongoDriverInformation == null ? MongoDriverInformation.builder() : MongoDriverInformation.builder(mongoDriverInformation))
+                .driverName("reactive-streams").build();
```
Comment (jyemin) [inline driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/MongoClients.java]: I think this fixes a regression introduce when reactive streams was pulled into the driver repository.  

Also, simplifies the driver name as described in the ticket

File: driver-legacy/src/main/com/mongodb/MongoClient.java (hunk 3 above)
Comment (rozza) [reply driver-legacy/src/main/com/mongodb/MongoClient.java]: Ack - good catch

**Pull request merged**